source/client/options_impl.cc

```diff
@@ -129,10 +130,19 @@ OptionsImpl::OptionsImpl(int argc, const char* const* argv) {
 
   TCLAP::ValueArg<std::string> tls_context(
       "", "tls-context",
-      "Tls context configuration in yaml or json. Example (json):"
+      "DEPRECATED, use --transport-socket instead. "
+      "Tls context configuration in yaml or json. Example (json): "
```

Comment:
Not sure of the proper place for this in the pull request, but is there any bizarre behavior if somebody provides both transport_socket and tls_context in the same request?  If so, should we fail up front if a user does that? Or alternatively, explicitly ignore tls_context?

Reply:
I'm adding a warning rather than an error to reduce effort. 

The command line test would become more complicated if we made them mutually exclusive. The cost of refactoring and reviewing the test could be too high considering:
- the low chance that someone specifies both and also misses the warning during the time --tls-context still exists
- the fact that the test would need to be refactored back to being simple when --tls-context is removed in the near future

Also, added a warning that --tls-context is deprecated.

Reply:
Perfect.  This works for me